rosidl_cmake/cmake/rosidl_convert_actions_to_msg_and_srv.cmake

```diff
@@ -0,0 +1,100 @@
+# Copyright 2018 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Convert action files from messages to services
+#
+# :param target: A target name to use to generate the .msg and .srv files
+# :type target: string
+# :param idl_files: A variable with a list of IDL files that may be actions.
+# :type target: list of strings
+# :param OUTPUT_IDL_VAR: Set to a list of message and service files
+# :type OUTPUT_IDL_VAR: list of strings
+#
+# @public
+#
+function(rosidl_convert_actions_to_msg_and_srv target)
+  cmake_parse_arguments(_ARG
+    ""
+    "OUTPUT_IDL_VAR"
+    ""
+    ${ARGN})
+  if(NOT _ARG_UNPARSED_ARGUMENTS)
+    message(FATAL_ERROR "rosidl_convert_actions_to_msg_and_srv() must be given IDL files")
+  endif()
+  if(NOT _ARG_OUTPUT_IDL_VAR)
+    message(FATAL_ERROR "rosidl_convert_actions_to_msg_and_srv() needs output variable for interfaces")
+  endif()
+
+  set(_action_files ${_ARG_UNPARSED_ARGUMENTS})
+
+  # Make a list of files that will be generated
+  set(_output_path "${CMAKE_CURRENT_BINARY_DIR}/action_msg_and_srv/${PROJECT_NAME}")
```

Comment:
@sloretz why not generating these messages below `${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}/${_parent_folder}` where `_parent_folder == 'action'`?

Reply:
In 18d40bb9d2dbae1bd896eb911d63ace6b7e2f43e changed `action_msg_and_srv` to the name of the package the code doing the conversion is in: `rosidl_actions`. This matches the C and C++ generators.